nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ListenPortDTO.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.web.api.dto;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.xml.bind.annotation.XmlType;
+
+import java.util.List;
+
+@XmlType(name = "listenPort")
+public class ListenPortDTO {
+
+    // Port definition
+    private String portName;
+    private int portNumber;
+    private String transportProtocol;
+    private List<String> applicationProtocols;
+
+    // Contextual information about the component providing the port, and the PG containing the component
+    private String componentType;
+    private String componentId;
+    private String componentName;
+    private String componentClass;
+    private String parentGroupId;
+    private String parentGroupName;
+
+    public String getPortName() {
+        return portName;
+    }
+
+    @Schema(description = "The name of the the listen port. Useful context for components that provide multiple ports.")
```

Comment:
It looks like this annotation should be on `getPortName()` instead of `setPortName()`